### PR TITLE
FRIN23-102: figure component

### PR DESCRIPTION
**Pull request by hasarindaKI**

### CODE REVIEW

File: app/models/concerns/katalyst/content/active_storage_extensions/has_one_image_attached.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                size:         {#{' '}
+                                less_than: Katalyst::Content.config.supported_max_image_size.megabytes,#{' '}
+                                message: "must be less than #{Katalyst::Content.config.supported_max_image_size} MB"#{' '}
+                }
+
+          def destroy_#{name}; end
+
+          def destroy_#{name}=(value)
+            self.#{name} = nil if value.present?
+          end
```
Comment (sobakasu) [inline app/models/concerns/katalyst/content/active_storage_extensions/has_one_image_attached.rb:27]: check if this is needed.

should be able to just do `#{name}=nil` without needing a destroy method

File: app/models/katalyst/content/figure.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        image.attach(source.image.blob) if source.respond_to?(:image) && source.image.attached?
+      end
+
+      def self.permitted_params
+        super + %i[image destroy_image caption alt_text]
+      end
+
+      def to_plain_text
+        [super, caption, alt_text].compact.join("\n") if visible?
+      end
```
Comment (sobakasu) [inline app/models/katalyst/content/figure.rb:26]: perhaps something like:

```
[super, "Image: #{alt_text}" if alt_text, "Caption: #{caption}" if caption].compact.join("\n")
```

Comment (hasarindaKI) [reply app/models/katalyst/content/figure.rb:23]: Updated

File: app/models/concerns/katalyst/content/active_storage_extensions/has_one_image_attached.rb (hunk 1 above)
Comment (hasarindaKI) [reply app/models/concerns/katalyst/content/active_storage_extensions/has_one_image_attached.rb]: Removed

**Pull request merged**